monitoring/promexp/main.go

```diff
@@ -58,6 +77,36 @@ func main() {
 </body></html>`))
 	})
 
+	// Forces a data push.
+	http.HandleFunc("/push", func(w http.ResponseWriter, r *http.Request) {
+		var msg string
+		if metrics, err := exporter.CollectRaw(); err == nil {
+			b := new(bytes.Buffer)
+			ts := time.Now().UnixNano()
+			for k, v := range metrics {
+				fmt.Fprintf(b, "%s value=%f %d\n", k, v, ts)
+			}
+			reqBody := b.String()
```

Comment:
Why convert to string if you convert it back to bytes a few lines below? Just `b.Bytes()`? Even that is probably not needed: just pass `b` instead of `bytes.NewReader([]byte(reqBody))`.  `b` is a Reader and a Writer.

See example here: https://play.golang.org/p/xbLrxC_rMQT

Reply:
Ah, nice!